modules/oe_subscriptions_anonymous/src/Plugin/ExtraField/Display/AnonymousSubscribeLink.php

```diff
@@ -59,36 +60,44 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function view(ContentEntityInterface $entity) {
     $build = [];
-    $entity_type = $entity->getEntityTypeId();
-    // Based on entity available flags.
-    $flags = $this->flag->getAllFlags($entity_type, $entity->bundle);
-    // No flaggings to subscribe through.
-    if (empty($flags)) {
+    $cache = new CacheableMetadata();
+    // Based on derivative id.
+    $flag = $this->flag->getFlagById($this->getDerivativeId());
+    // No flag empty return.
```

Comment:
A good comment would be "Bail out if the flag cannot be loaded."

Reply:
Changed